add moveLiveRegion function to keep live regions within <md-autocomplete>

0ddfell0w opened the pull request:
Fixes https://github.com/angular/material/issues/10804

Adds a function (called on init) to the autocompleteController.js that moves the component's live region to be inside the md-autocomplete element. This fixes a bug where autocompletes within dialogs were hidden with aria-hidden="true", since they were not a direct ancestor of the dialog.

The fix works because it guarantees that aria-hidden will not be applied to the live region for the autocomplete being created within the dialog.

katiehockman commented on src/components/autocomplete/js/autocompleteController.js:
```diff
@@ -196,6 +197,16 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   /**
+   * Moves the live region within the autocomplete to ensure the live region is
+   * is not hidden from screen readers when the autocomplete appears within
+   * a dialog
```
Add a period here :dancer:

katiehockman commented on src/components/autocomplete/js/autocompleteController.js:
```diff
@@ -196,6 +197,16 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
 
   /**
+   * Moves the live region within the autocomplete to ensure the live region is
+   * is not hidden from screen readers when the autocomplete appears within
```
s/screen readers/assistive technology

katiehockman commented:
I tested on your local instance with Voiceover, and the changes look good! Thanks for fixing this, it's such an odd bug. :eagle:

0ddfell0w committed:
fix(dialog):prevent dialog from hiding live regions on open

change how dialogs hide background elements to avoid hiding live
regions, ensuring that an autocomplete inside a dialog will not have
its live region aria-hidden. Adds unit test

Fixes #10804
